client/my-sites/pages/page/index.js

```diff
@@ -203,6 +204,23 @@ class Page extends Component {
 		);
 	}
 
+	getFrontPageItem() {
+		if ( this.props.hasStaticFrontPage && this.props.isPostsPage ) {
+			return null;
+		}
+
+		if ( ! utils.userCan( 'edit_post', this.props.page ) ) {
+			return null;
+		}
+
+		return (
+			<PopoverMenuItem onClick={ this.setFrontPage }>
```

Comment:
Even though it won't be fully functional yet, clicking on this menu item should still do _something_.  If you make a `setFrontPage() {` method, you can call `alert` from there and show a message indicating that this feature is still in development.